tools/outline_proxy_controller/outline_error.h

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// This file contains the C++11 standard conforming implementation of
+// all errors used by outline.
+
+#pragma once
+
+#include <system_error>
+
+namespace outline {
+
+/**
+ * @brief The standard error code constants used by outline.
+ * @remarks The codes are copied from "/src/www/model/errors.ts".
+ */
+enum class ErrorCode {
+  kOk = 0,
+  kUnexpected = 1,
+  kVpnPermissionDenied = 2,
+  kInvalidServerCredentials = 3,
+  kUdpRelayNotEnabled = 4,
+  kServerUnreachable = 5,
+  kVpnStartFailure = 6,
```

Comment:
Not for this PR, but we need better errors. We need to know why the VPN failed to start.

Reply:
Yup, also I think we can enable sentry reports in our service as well.